### PROBLEM STATEMENT
We would be performing logistic regression on a data set gotten from UCI machine learning repository to classify breast
cancer as either benign or malignant. 

A clean version of this data set is used (No missing values) so no data preprocessing needed to clean the data.

### IMPORTING THE LIBRARIES

In [97]:
import pandas as pd

### IMPORTING THE DATASET

In [98]:
dataset=pd.read_csv('breast_cancer.csv')

### CREATING THE SET OF FEATURES(INDEPENDENT VARIABLES) AND THE DEPENDENT VARIABLE

In [99]:
x=dataset.iloc[:,1:-1].values
y=dataset.iloc[:,-1].values

### SPLITTING INTO TRAINING SET AND TEST SET

In [100]:
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test=train_test_split(x,y,test_size=0.2, random_state=0)

### FITTING THE LOGISTIC REGRESSION MODEL ON THE TRAINING SET

In [101]:
from sklearn.linear_model import LogisticRegression
classifier=LogisticRegression(random_state=0)
classifier.fit(x_train,y_train)

LogisticRegression(random_state=0)

### PREDICTING THE RESULT

In [102]:
y_pred=classifier.predict(x_test)

### MAKING THE CONFUSION MATRIX

In [103]:
from sklearn.metrics import confusion_matrix
cm=confusion_matrix(y_test,y_pred)
cm

array([[84,  3],
       [ 3, 47]], dtype=int64)

### COMPUTING THE ACCURACY WITH K FOLD CROSS VALIDATION
We are using this method to get a relevant measure of the accuracy of our logistic regression model

In [104]:
from sklearn.model_selection import cross_val_score
accuracies=cross_val_score(estimator=classifier,X=x_train, y=y_train, cv=10)
print('Accurcay: {:.2f}%'.format(accuracies.mean()*100))
print('Standard variation: {:.2f}%'.format(accuracies.std()*100))

Accurcay: 96.70%
Standard variation: 1.97%


### SUMMARY
The Accuracy of our model is 96.70%.

The standard variation is 1.97% meaning the accuracy of each fold since we are using 10 folds, fall just within + or - 1.97 
of our accuracy score which is 96.70%

The logistic regression model is obiviosuly performing well on our dataset and could definitely be used by doctors in their 
classification of breast cancer cases